src/pages/index.tsx

```diff
@@ -206,6 +212,7 @@ export default function IndexPage() {
                       <ProposalItem
                         key={proposal.id}
                         proposal={proposal}
+                        coauthorRequest={!!pendingCoauthorRequests.find((req) => req.proposal_id === proposal.id)}
```

Comment:
If you aren't passing the entire request object as prop and `coauthorRequest` is a boolean I'd call this `hasCoauthorRequest`